solutions/usaco-809.mdx

```diff
@@ -2,14 +2,97 @@
 id: usaco-809
 source: USACO Bronze 2018 Febuary
 title: Taming the Herd
-author: Sathvik Chundru
+author: Sathvik Chundru, Aditya Gupta
 ---
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_taming_bronze_feb18.html)
 
 **Time Complexity:** $\mathcal{O}(N)$
 
 <LanguageSection>
+	
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int main() {
+	freopen("taming.in", "r", stdin);
+	freopen("taming.out", "w", stdout);
+	int n;
+	cin >> n;
+	/*
+	 * A status of 2 means a breakout happened on day n.
+	 * A status of 1 means a breakout did not happen on day n.
+	 * A status of 0 means we are not sure what happened on day n.
+	 */
+	vector<int> status(n, 0);
+	status[0] == 2;  // a breakout must occur on the first day
+	vector<int> log(n);  // log for each day
+	for (int i = 0; i < n; i++) {
+		cin >> log[i];
+	}
+	bool valid = true;
+	if (log[0] == 1) {
+		valid = false; // we found a discrepancy
+	} else {
+		status[0] = 2; // the problem says the first day had a breakout
+	}
+	
+	// we go through our log, updating our status array accordingly
+	for (int i = 1; i < n; i++) {
+		if (!valid) break;
+		if (log[i] == -1) {
+	        	continue; // we have no log for this day, so we continue to the next day
+        	} else if (log[i] == 0) {
+			status[i] = 2; // this means that we have a breakout on this day
+        	} else {
+			// we now process the log[i] days, checking for discrepancies along the way
+			status[i] = 1; // the current day does not have a breakout
+			int curr_index = i - 1;
+			log[i]--;
```

Comment:
explain this whole loop. How does these conditions work?

Reply:
actually never mind it’s kind of self explanatory lol

Reply:
Yea, I was about to say, lol